### ISSUE TITLE
const vars declared in different scopes conflict each other 

### ISSUE THREAD

**Issue by lazutkin** [jshint/jshint#1884]
`const`s  with same name but declared in different scopes are in conflict with each other

```javascript
function t( ){
  'use strict';
  const x = 1;

  if( true) {
    const x = 2;
  }
}
```
```
error  11| const 'x' has already been declared.
warning   4| 'x' is already defined.
```

**Comment by lukeapage** [comment]
@jugglinmike this can be closed now

**Comment by jugglinmike** [comment]
Confirmed--resolved via gh-2326 / 1bb80f9a5551a639db2be0dea655c9edef7a9b8a . Expect this in the next release of JSHint (version > 2.7.0).
